gr-qtgui/lib/CMakeLists.txt

```diff
@@ -87,10 +101,25 @@ target_include_directories(
     PUBLIC $<INSTALL_INTERFACE:include>
            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
     PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
-target_link_libraries(gnuradio-qtgui
-    PUBLIC ${QTGUI_LIBS}
-    PRIVATE ${PRIVATE_LIBS}
-)
+# Windows and Mac need different linkage for OpenGL functions
+if(UNIX AND NOT APPLE OR NOT ENABLE_GR_QTGUI_OPENGL)
+    target_link_libraries(gnuradio-qtgui
+        PUBLIC ${QTGUI_LIBS}
+        PRIVATE ${PRIVATE_LIBS}
+    )
+elseif(APPLE)
+    target_link_libraries(gnuradio-qtgui
+        PUBLIC ${QTGUI_LIBS}
+        PRIVATE ${PRIVATE_LIBS}
+        PRIVATE "-framework OpenGL"
+    )
+else()
+    target_link_libraries(gnuradio-qtgui
+        PUBLIC ${QTGUI_LIBS}
+        PRIVATE ${PRIVATE_LIBS}
+        PRIVATE opengl32
+    )
+endif()
```

Comment:
Rather than linking the specific OpenGL library/framework for each platform, I think the best way would be to `find_package(OpenGL ...)` when we have `ENABLE_GR_QTGUI_OPENGL` (this could be part of the if statement on line 73) and then add the provided `OpenGL::GL` target to `PRIVATE_LIBS`.

I don't know why this is necessary for macOS and Windows but not Linux. Ideally I'd think that the `Qt5::OpenGL` target would bring along the necessary GL libraries if it requires them. Then again, this block directly calls `gl*` functions and so maybe it *is* proper to have to specifically link with the GL libraries (so why does Linux get away with not doing it?). Anyway, unless someone knows better, the goal is to just get it building, so these links are evidently necessary.

Reply:
Thanks, that is much cleaner